studio/stores/UiStore.ts

```diff
@@ -6,7 +6,7 @@ import Telemetry from 'lib/telemetry'
 
 export interface IUiStore {
   language: 'en_US'
-  theme: 'dark' | 'light'
+  theme: 'dark' | 'light' | 'system'
```

Comment:
Currently the studio app only supports `dark` and `light` theme. We don't have `system` theme. I believe there's a miss-understanding here.

Reply:
According to https://github.com/supabase/supabase/issues/4329:
- adding a new `system` selection option on User preference page (DONE)
- checking window.localStorage.getItem('theme')
  - if `system`: Then we detect if the system is dark mode or not to set the theme value `dark` or `light`.
  - if `dark` or `light`: set theme value according to

Reply:
Do you want me to check for `system` in `ui.load` method, and then accordingly set the theme to `light` or `dark`?

Reply:
- if `system`: Then we detect if the system is in dark mode or not to set the theme value `dark` or `light`.
- if `dark` or `light`: set theme value according to

I have already added a check for this in the [`ui.setTheme`](https://github.com/supabase/supabase/pull/4482/files#diff-39dba4c513e1bf9dcffbb0250c784c1b88d1543fcc8e97a074132bf39be31cddL88-R98) method. If you want `ui.setTheme()` to only have `light` or `dark` as props.

Reply:
Let wait for @joshenlim confirmation. 

To me you only need to update load() method under UiStore to detect `theme` value. Your load method looks ok to me except 
```
load() {
  if (typeof window === "undefined") return;
  const savedTheme = (window.localStorage.getItem("theme") ?? "dark") as
    | "dark"
    | "light"
    | "system";
  if (savedTheme === "dark" || savedTheme === "light") {
   // why do you want to save theme value here again? 
    window.localStorage.setItem("theme", savedTheme);
    return this.setTheme(savedTheme);
  }
  if (savedTheme === "system") {
   // same question
    window.localStorage.setItem("theme", "system");
    return this.setTheme(
      window.matchMedia("(prefers-color-scheme: dark)").matches ? "dark" : "light"
    );
  }
  // same question
  window.localStorage.setItem("theme", "dark");
  this.setTheme("dark");
};
```

The rest `toggleTheme`, `setTheme `, `isDarkTheme ` should be kept as it is

Reply:
Let clarify the term:
- theme-option: 3 dark, light and system
- theme: only 2 dark and light

Reply:
Sorry for that, I don't need to update the theme value in `localStorage` for every case. The value only needs to be updated if value received is not `light`, `dark` or `system`.

Reply:
@phamhieu If `ui.setTheme` method is suppose to have only `light` or `dark` as props, the method will be as follows. This is because the theme option in `localStorage` should be updated every time the `ui.seTheme()` is called outside the `ui.load`.

```ts
setTheme(theme: "dark" | "light", asSystem?: boolean) {
  if (asSystem) {
    this.theme = theme
    window.localStorage.setItem("theme", "system")
    if (theme === "light") return (document.body.className = "light")
    return (document.body.className = "dark")
  }
  this.theme = theme
  window.localStorage.setItem("theme", theme)
  document.body.className = theme
}
```

I still think using `system` as prop is more readable. This removes the need to fetch the `localStorage` every time we need to check if the theme option is set to `system` or not. It also removes the need for the `asSystem` prop. I will wait for the final decision and make the requested changes.

Reply:
setTheme should not care about `theme-option` at all. 

When users select the system option, you can check the system config first then call UiStore.SetTheme with the final `theme` value.

Reply:
The theme options are `light`, `dark` and `system`. When the `system` theme-option is selected, we will pass the current system scheme (eg. `dark`) to the `setTheme` method and set the theme as `dark`, but this will also set the theme-option as `dark` and change the `localStorage` value to `dark`. So the next time the browser opens the theme will be inferred as `dark` and not `system`. Is this is what you mean?

Reply:
> and change the localStorage value to dark

We save `theme-option` to local-storage NOT `theme`. And we already update `load()` method to support it.

Reply:
I just check the current code for `setTheme`.  I'm so sorry for the confusion. Just to reconfirm
- we have to remove the save to localstorage inside `setTheme` and move it to the theme option selection page
- inside setTheme we 
https://github.com/supabase/supabase/blob/305fce8a019e89269c3d6152842c2ab5180e13c7/studio/stores/UiStore.ts#L88-L92

Reply:
So the only place where the theme-option will be stored is `localStorage`? And the `setTheme` method will not update the theme-option in `localStorage`, but will be done by functions that call `setTheme`?